### PR TITLE
Helix upgrade to 0.6.7.

### PR DESCRIPTION
**Pull request by htran1**
Created customized Helix TaskDriver, MessagingService, and CriteriaEvaluator classes to workaround bug, missing functionality, and behavior changes.
Issues that we had to workaround.
1. Messaging service does not allow sending messages to instances
2. Deletion of job is not allowed on a running queue
3. Cannot stop a queue
4. Shutdown message is not defined
5. Creation of queue fails with generic exception if queue already exists

### CODE REVIEW

**Comment by htran1** [comment]
@sdas, @abti, please take a look.

**Comment by htran1** [comment]
@kishoreg, please take a look.

File: gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobLauncher.java [hunk 1]
```diff
@@ -147,42 +141,6 @@ public GobblinHelixJobLauncher(Properties jobProps, final HelixManager helixMana
...
-               }
-               **/
-
-              // HACK to ensure that concurrent tasks per instance is set.
-              // TODO: Remove this when we upgrade to a fixed Helix version
-              final String taskResourceName = TaskUtil.getNamespacedJobName(jobContext.getJobName(), jobContext.getJobId());
-              int jobConcurrency = ConfigUtils.getInt(jobConfig, GobblinClusterConfigurationKeys.HELIX_CLUSTER_TASK_CONCURRENCY,
-                  GobblinClusterConfigurationKeys.HELIX_CLUSTER_TASK_CONCURRENCY_DEFAULT);
-              HelixConfigScope resourceScope =
-                  new HelixConfigScopeBuilder(HelixConfigScope.ConfigScopeProperty.RESOURCE).forCluster(clusterName).forResource(taskResourceName).build();
```
Comment (shirshanka) [inline gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobLauncher.java]: When we delete this code, do we know if concurrent jobs per instance setting will be honored?

File: gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixMessagingService.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+  @Override
+  public Map<InstanceType, List<Message>> generateMessage(final Criteria recipientCriteria,
+      final Message message) {
+    Map<InstanceType, List<Message>> messagesToSendMap = new HashMap<InstanceType, List<Message>>();
+    InstanceType instanceType = recipientCriteria.getRecipientInstanceType();
+
+    if (instanceType == InstanceType.CONTROLLER) {
+      List<Message> messages = generateMessagesForController(message);
+      messagesToSendMap.put(InstanceType.CONTROLLER, messages);
+      // _dataAccessor.setControllerProperty(PropertyType.MESSAGES,
```
Comment (shirshanka) [inline gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixMessagingService.java]: remove commented code?

File: gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobLauncher.java (hunk 1 above)
Comment (htran1) [inline gobblin-cluster/src/main/java/gobblin/cluster/GobblinHelixJobLauncher.java]: I checked and confirmed that it is now set to 100, so the call jobConfigBuilder.setNumConcurrentTasksPerInstance(100); is now working.
Here is the job config from zookeeper showing the value of ConcurrentTasksPerInstance.
"simpleFields":{ "Command":"GobblinTaskFactory" ,"ConcurrentTasksPerInstance":"100"

**Pull request merged**